File: LeetCode/Easy/100_Same Tree.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _100_Same_Tree
{
    class Program
    {
        static void Main(string[] args)
        {
            Solution solution = new Solution();
            TreeNode p = new TreeNode(1)
            {
                left = new TreeNode(2)
            };
            TreeNode q = new TreeNode(1)
            {
                right = new TreeNode(2)
            };

            Console.WriteLine(solution.IsSameTree(p, q));
            Console.ReadLine();
        }

        //Definition for a binary tree node.
        public class TreeNode
        {
            public int val;
            public TreeNode left;
            public TreeNode right;
            public TreeNode(int x) { val = x; }
        }

        public class Solution
        {
            public bool IsSameTree(TreeNode p, TreeNode q)
            {
                return (p == null && q == null) ||
                    (p != null && q != null && p.val == q.val 
                    && IsSameTree(p.left, q.left) && IsSameTree(p.right, q.right));
            }
        }
    }
}

File: LeetCode/100_Same Tree.py
# Definition for a binary tree node.


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def isSameTree(self, p: TreeNode, q: TreeNode) -> bool:
        # 如果兩個節點都是None，則認為它們相同
        if not p and not q:
            return True
        # 如果其中一個節點是None而另一個不是，則它們不相同
        if not p or not q:
            return False
        # 如果節點的值不相等，則它們不相同
        if p.val != q.val:
            return False
        # 遞迴檢查左子樹和右子樹
        return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)
        